projects/core/src/site-context/facade/base-site.service.ts

```diff
@@ -23,21 +23,29 @@ export class BaseSiteService implements SiteContext<string> {
    */
   getActive(): Observable<string> {
     return this.store.pipe(
-      select(SiteContextSelectors.getActiveBaseSite),
+      select(SiteContextSelectors.getActiveBaseSiteUid),
       filter((active) => Boolean(active))
     );
   }
 
   /**
    * We currently don't support switching baseSite at run time
    */
-  getAll(): Observable<string[]> {
-    return this.getActive().pipe(map((baseSite) => [baseSite]));
+  getAll(): Observable<BaseSite[]> {
```

Comment:
I'm afraid this is quit unexpected for customers. I'd prefer gradually moving away from the string array. Can you come up with a cleaner way so?

Reply:
We replace `string[]` to `BaseSite[]`, and will handle breaking change in #9601